kore/src/Kore/Exec.hs

```diff
@@ -277,33 +284,47 @@ trackExecDepth transit prim (execDepth, execState) = do
 
 -- | Project the value of the exit cell, if it is present.
 getExitCode
-    :: (MonadIO simplifier, MonadSimplify simplifier)
+    :: forall simplifier
+    .  (MonadIO simplifier, MonadSimplify simplifier)
     => VerifiedModule StepperAttributes
     -- ^ The main module
-    -> Pattern VariableName
+    -> OrPattern.OrPattern VariableName
     -- ^ The final configuration(s) of execution
     -> simplifier ExitCode
 getExitCode indexedModule finalConfig =
     takeExitCode $ \mkExitCodeSymbol -> do
         let mkGetExitCode t = mkApplySymbol (mkExitCodeSymbol []) [t]
-        exitCodePattern <-
-            Pattern.simplifyTopConfiguration (mkGetExitCode <$> finalConfig)
-            >>= Logic.scatter
-        case Pattern.term exitCodePattern of
-            Builtin_ (Domain.BuiltinInt (Domain.InternalInt _ exit))
-              | exit == 0 -> return ExitSuccess
-              | otherwise -> return $ ExitFailure $ fromInteger exit
-            _ -> return $ ExitFailure 111
+        exitCodePatterns <-
+            do
+                pat <- Logic.scatter $ fmap mkGetExitCode `MultiOr.map` finalConfig
+                Pattern.simplifyTopConfiguration pat
+            & MultiOr.gather
+        let flatExitCodePatterns = MultiOr.flatten exitCodePatterns
+            exitCode =
+                case toList (MultiOr.map Pattern.term flatExitCodePatterns) of
+                    [exitTerm] -> extractExit exitTerm
+                    _      -> ExitFailure 111
+        return exitCode
```

Comment:
```suggestion
        exitCodePatterns <-
            do
                config <- Logic.scatter configs
                Pattern.simplifyTopConfiguration (mkGetExitCode <$> config)
                    >>= Logic.scatter
            & MultiOr.gather
        let exitCode =
                case toList (MultiOr.map Pattern.term exitCodePatterns) of
                    [exitTerm] -> extractExit exitTerm
                    _      -> ExitFailure 111
        return exitCode
```

Reply:
@ttuegel If I use `gather :: (Ord a, TopBottom a, MonadLogic m) => m a -> m (MultiOr a)` there, then the lambda function given to `takeExitCode` has to return a `MonadLogic` result. Then how can I change its signature to
`takeExitCode :: (([Sort] -> Symbol) -> simplifier ExitCode)`?

Reply:
Sorry, I meant to suggest using `MultiOr.observeAllT` instead of `MultiOr.gather`.